BaseWeapon/src/main/java/com/github/sef24sp4/baseweapon/BaseProjectile.java

```diff
@@ -0,0 +1,29 @@
+package com.github.sef24sp4.baseweapon;
+
+import com.github.sef24sp4.common.entities.IAttackingEntity;
+import com.github.sef24sp4.common.entities.ICollidableEntity;
+import com.github.sef24sp4.common.interfaces.IEntityManager;
+import com.github.sef24sp4.common.metadata.GameElementType;
+import com.github.sef24sp4.common.projectile.CommonProjectile;
+
+public class BaseProjectile extends CommonProjectile implements ICollidableEntity, IAttackingEntity {
+	private final double damage = 1;
+	protected BaseProjectile(ICollidableEntity shooter) {
+		super(shooter);
+	}
+
+
+
+	@Override
+	public void collide(IEntityManager entityManager, ICollidableEntity otherEntity) {
```

Comment:
Please also make a check to prevent the projectile from being removed when it collides with the shooter.